javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -165,8 +164,8 @@ private Vector(BitMappedTrie<T> trie) {
             final Object[] array = ((Collection<? extends T>) iterable).toArray();
             return ofAll(BitMappedTrie.ofAll(array));
         } else {
-            final Seq<? extends T> elems = seq(iterable);
-            final Object[] array = asArray(elems.iterator(), elems.size());
+            final Seq<? extends T> result = (iterable instanceof Seq) ? (Seq<T>) iterable : List.ofAll(iterable);
+            final Object[] array = asArray(result.iterator(), result.size());
```

Comment:
We create a Seq to get the size. That's a waste (of mem and cpu/gc) because several new objects are created. Instead I would create two times an iterator(), the first for counting, the second for getting the elements:

```java
int size = 0;
for (Iterator<?> iter = iterable.iterator(); iter.hasNext(); iter.next()) { size += 1; }
final Object[] array = asArray(iterable.iterator(), size);
```

Additionally we could also do an extra check for `Traversable`:

```java
} else if (iterable instanceof Traversable<?>) {
    final Traverable<T> traversable = (Traversable<T>) iterable;
    final Object[] array = asArray(traversable.iterator(), traversable.size());
    return ofAll(BitMappedTrie.ofAll(array));
} ...
```